### DISCUSSION TITLE
Transmitting Microphone Signal to Phone

### DISCUSSION THREAD

**Discussion by syedadil19** [pschatzmann/ESP32-A2DP#499] [Q&A] [answered]
Hello, I am in the middle of a project which requires me to transmit an audio signal from an analog microphone to a phone. I have a list of questions I've been dreading to have answers to.

1. I've been using an ESP32 to get used to this A2DP library. However, my team preferred to use Arduino Nano 33 BLE. Based on my understanding, the Nano 33 BLE does not support I2S and A2DP right? Or is there a way to get around it? We require a very small Arduino board for our project. I've also checked out Arduino Nano ESP32 but I'm not quite brave enough to suggest this to my teammates yet without some supporting documentation. I've checked that the Nano ESP32 supports both I2S and A2DP which means that this library and all other ESP32 codes should work well with it right?

2. I've also looked through this example of yours, https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-basic-api/base-adc-a2dp. But, I can't find the device's name on my phone which means I can't connect my phone to the device. I've had  ArduinoBLE, your A2DP and Audio Tools libraries installed and confirmed their installations. I am using an iPhone 13 Pro and this is the serial output data:
rst:0x3 (SW_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee

And how do I confirm that my phone indeed has received the audio signal spoken into the microphone?
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1344
load:0x40078000,len:13964
load:0x40080400,len:3600
entry 0x400805f0

I apologize for the long questions but I would really appreciate anyone's answer to my questions. Thank you all in advance.

**Comment by pschatzmann** [comment, marked as answer]
1) This A2DP library works only on the ESP32. So if you want to use the Nano 33 BLE you will need to use WiFi or BLE
I suggest to have a look at the communications examples of the AudioTools: tcp, udp or vban might be of interrest.
The Nano ESP32 is based on an ESP32-S3, which does not support classic BlueTooth, so there is no A2DP as well.
ps, I have spend quite some time to have I2S working on the Nano 33 BLE.

2) This will not work: a mobile phone is not an A2DP Sink (=Bluetooth Speaker)!

**Reply by syedadil19** [reply]
Do you think this link https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-stream/streams-adc-serial would be useful for me?

**Reply by pschatzmann** [reply]
Sure, to test if you get some valid audio data.
For receiving and transmitting: its https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-communication

**Reply by syedadil19** [reply]
It looks like all the examples within this link uses WiFi connectivity. We planned for a wireless connection that can maintain connectivity without the WiFi network.

**Comment by pschatzmann** [comment]
If you need to rely on BLE you are in a pretty bad spot: I am not aware of any open source implementation that would support proper BLE Audio (on microcontrollers).

It will be quite a challange to get a working solution in the light of the low transmission speeds and MTUs that are available. You will certainly need to use a good codec that does compression.

**Reply by syedadil19** [reply]
I see. Thank you for your concern. One more question for now, does the Nano 33 BLE support Bluetooth Classic as well? I’ve been exploring but can’t come to a definitive conclusion. I hope you can provide some explanations. Thank you.

**Comment by pschatzmann** [comment]
I was working on a [BLE based prototype](https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/sandbox/ble) and I was positively surprised that I have it working with a sample rate of 44100 on 2 channels.

I have it implemented for an ESP32, but unfortunately the API varies from platform to platform.
Do you see any chance that you can contribute the implementation for the Nano 33 BLE ?

**Reply by syedadil19** [reply]
I'll see what I can do and let you know soon.

I was trying to upload your code "communications-ble-server-send.ino" to an ESP32 but received the following error,

Sketch uses 1337549 bytes (102%) of program storage space. Maximum is 1310720 bytes.
Global variables use 46904 bytes (14%) of dynamic memory, leaving 280776 bytes for local variables. Maximum is 327680 bytes.
Sketch too big; see https://support.arduino.cc/hc/en-us/articles/360013825179 for tips on reducing it.
text section exceeds available space in board

Compilation error: text section exceeds available space in board

What seems to be the issue here? I've tried on another laptop which gave me the same error.

**Reply by syedadil19** [reply]
Hi Phil, the error has been rectified and it’s no longer bothering me. I was able to upload your code to my ESP32 to test it out. But, how did you confirm the transmission of the audio? I was using the nRF Connect for Mobile App and was only able to view the output as a hex string. Mind sharing your method with me?
